src/main/scala/io/iohk/ethereum/validators/SignedTransactionValidator.scala

```diff
@@ -83,11 +90,72 @@ object SignedTransactionValidator extends SignedTransactionValidator {
+    *
+    * @param stx Transaction to validate
+    * @param blockHeaderNumber Number of the block where the stx transaction was included
+    * @return Either the validated transaction or a TransactionNotEnoughGasForIntrinsicError
+    */
+  private def validateGasLimitEnoughForIntrinsicGas(stx: SignedTransaction, blockHeaderNumber: BigInt): Either[SignedTransactionError, SignedTransaction] = {
+    import stx.tx
+    val config = EvmConfig.forBlock(blockHeaderNumber, blockchainConfig)
+    val txIntrinsicGas = config.calcTransactionIntrinsicGas(tx.payload, tx.isContractInit)
+    if (stx.tx.gasLimit >= txIntrinsicGas) Right(stx)
+    else Left(TransactionNotEnoughGasForIntrinsicError(s"Tx gas limit ${stx.tx.gasLimit} < tx intrinsic gas $txIntrinsicGas"))
+  }
+
+  /**
+    * Validates the sender account balance contains at least the cost required in up-front payment.
+    *
+    * @param stx Transaction to validate
+    * @param senderBalance Balance of the sender of the tx
+    * @param calculateUpfrontCost Function used to calculate the upfront cost of the transaction tx
+    * @return Either the validated transaction or a TransactionSenderCantPayUpfrontCostError
+    */
+  private def validateAccountHasEnoughGasToPayUpfrontCost(stx: SignedTransaction, senderBalance: UInt256, calculateUpfrontCost: Transaction => UInt256)
+  : Either[SignedTransactionError, SignedTransaction] = {
+    val upfrontCost = calculateUpfrontCost(stx.tx)
+    if (senderBalance >= upfrontCost) Right(stx)
+    else Left(TransactionSenderCantPayUpfrontCostError(s"Upfrontcost ($upfrontCost) > sender balance ($senderBalance)"))
+  }
+
+  /**
+    * The sum of the transaction’s gas limit and the gas utilised in this block prior must be no greater than the
+    * block’s gasLimit
+    *
+    * @param stx           Transaction to validate
+    * @param accumGasUsed Gas spent within tx container block prior executing stx
+    * @param blockGasLimit Block gas limit
+    * @return Either the validated transaction or a TransactionGasLimitTooBigError
+    */
+  private def validateBlockHasEnoughGasLimitForTx(stx: SignedTransaction, accumGasUsed: BigInt, blockGasLimit: BigInt)
+  : Either[SignedTransactionError, SignedTransaction] = {
+    if (stx.tx.gasLimit + accumGasUsed <= blockGasLimit) Right(stx)
+    else Left(TransactionGasLimitTooBigError(s"Tx gas limit (${stx.tx.gasLimit}) + gas accum (${accumGasUsed}) > block gas limit ($blockGasLimit)"))
+  }
 }
 
 sealed trait SignedTransactionError
 
 object SignedTransactionError {
   case object TransactionSignatureError extends SignedTransactionError
-  case class TransactionSyntaxError(msg: String) extends SignedTransactionError
+  case class TransactionSyntaxError(reason: String) extends SignedTransactionError
+  case class TransactionNonceError(reason: String) extends SignedTransactionError
+  case class TransactionNotEnoughGasForIntrinsicError(reason: String) extends SignedTransactionError
```

Comment:
mmm...maybe this error message can be constructed based on `stx.tx.gasLimit` and `txIntrinsicGas` and not received as parameter